model.go

```diff
@@ -1295,6 +1311,16 @@ func newModelFromValid(valid map[string]interface{}, importVersion int) (*model,
 	}
 	result.setRemoteApplications(remoteApplications)
 
+	// environ-version was added in version 3. For older schema versions,
+	// the environ-version will be set to the zero value.
+	if importVersion >= 3 {
+		num, err := version.Parse(valid["environ-version"].(string))
```

Comment:
Good question. Originally it held the controller version, but it's not independent, and could just as well be an integer. I'll change it to an int when I'm back at work tomorrow.